digitalpalireader/content/js/dpr_pal.js

```diff
@@ -133,15 +133,38 @@ console.log('Loading DPR_PAL...');
     }
   }
 
+  /*
+  if 'copy permalink to clipboard' button on the modal is pressed , the focus is on the modal .
+  thus execCommand("copy") does not copy from the element appended to the body .
+
+  one workaround is to append an element to the modal itself when the modal is open.
+
+  refer : https://stackoverflow.com/questions/48866903/copy-to-clipboard-action-doesnt-work-when-modal-is-open
+  */
+
   DPR_PAL.copyToClipboard = text => {
+    SourceisModal = document.getElementsByClassName("modal-open").length;
```

Comment:
1. please move this inside the isWeb check.
2. remove the 2 if(sourceismodals) cases and instead get the right element in the declaration itself